components/layout_thread/lib.rs

```diff
@@ -934,6 +934,14 @@ impl LayoutThread {
                         let origin = Rect::new(Point2D::new(Au(0), Au(0)), root_size);
                         build_state.root_stacking_context.bounds = origin;
                         build_state.root_stacking_context.overflow = origin;
+
+                        if !build_state.iframe_sizes.is_empty() {
+                            let msg = ConstellationMsg::FrameSizes(build_state.iframe_sizes.clone());
```

Comment:
nit: Since the iframe sizes are not going to be useful, you can prevent the extra allocation doing the following:

```rust
let iframe_sizes = mem::replace(&mut build_state.iframe_sizes, vec![]);
let msg = ConstellationMsg::FrameSizes(iframe_sizes);
```

You can also make that first line a method in `DisplayListBuildState` if you want, something along the lines of:

```rust
fn take_iframe_sizes(&mut self) -> Vec<..> {
    mem::replace(..)
}
```

But given it's only used here, it's not a big deal.